File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.go
func minSubArrayLen(target int, nums []int) int {
    sum := 0
    left := 0
    right := 0
    numsLenght := len(nums)
    resLenght := 1000001
        
    for  right < numsLenght {
            
        if sum < target {
            sum += nums[right]
            right++
                
        } else{
            resLenght = min(resLenght, right - left);
            sum -= nums[left]
            left++
        }
            
    }
        for  sum >= target {
            resLenght = min(resLenght, right - left)
            sum -= nums[left]
            left++
        }
        
    if(resLenght == 1000001) {
        return 0
    }
    return resLenght
}

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int sum = 0;
        
        int left = 0;
        int right = 0;
        int numsLenght = nums.size();
        int resLenght = 1000000;
        while(right < numsLenght){
            
            if(sum < target){
                sum += nums[right];
                right++;
                
            }
            else{
                resLenght = min(resLenght, right - left);
                sum -= nums[left];
                left++;
            }
            
        }
        while(sum >= target){
            resLenght = min(resLenght, right - left);
            sum -= nums[left];
            left ++;
        }
        if(resLenght == 1000000) return 0;
        return resLenght;
    }
};